fatiando/gravmag/harvester.py

```diff
@@ -221,7 +221,7 @@ def _find_index(point, mesh):
 
 
 def harvest(data, seeds, mesh, compactness, threshold, report=False,
-            restrict=None):
+            restrict=[None]):
```

Comment:
@cmeessen it's really dangerous to use mutable python types (lists, dicts, etc)
as default function values. These are evaluated at import time and can then be
changed between successive calls to the function, for example:

    def func(y=[]):
        if y:
            return 'not empty'
        else:
            y.append('anything')
            return 'empty'


    >>> func()
    'empty'
    >>> func()
    'not empty'

That's why we always use `Ǹone` instead of an empty list or dict, even when the
required argument type is one of those.

I had a really hard to find bug in `fatiando.mesher` because I was setting
`props={}` as default. Values were changing unexpectedly and I had no idea why.

So you can leave this as `restrict=None`.